trace/config.go

```diff
@@ -229,6 +247,17 @@ func WithTimestamp(t time.Time) SpanEventOption {
 	return timestampOption(t)
 }
 
+type stackTraceOption bool
+
+func (o stackTraceOption) applyEvent(c *EventConfig)  { c.stackTrace = bool(o) }
+func (o stackTraceOption) applySpan(c *SpanConfig)    { c.stackTrace = bool(o) }
+func (o stackTraceOption) applySpanEnd(c *SpanConfig) { o.applySpan(c) }
+
+// WithStackTrace sets the flag to capture the error with stack trace (e.g. true, false).
+func WithStackTrace(b bool) SpanEndEventOption {
+	return stackTraceOption(b)
+}
```

Comment:
does the `b bool` parameter make any sense here? how about simplifying (and maybe renaming) it to

```suggestion
// WithStackTraceOnRecordError captures the stack trace for error events.
func WithStackTraceOnRecordError() SpanEndEventOption {
	return stackTraceOption(true)
}
```

Reply:
`WithStackTraceOnRecordError` implies this can only be used for the `RecordError` method which is not the case nor is it something that I think we want to restrict this to.

Reply:
> @MrAlias then maybe just removing the `b bool` parameter?

SGTM. I know @Aneurysm9 prefers this parameter so any loop setting options from other variables only needs to pass an argument instead of including a one-off conditional.

Reply:
> I know @Aneurysm9 prefers this parameter so any loop setting options from other variables only needs to pass an argument instead of including a one-off conditional.

Yup, I've been bitten by this before and would prefer to always have the option to include it.  If we want to get clever we could do this:

```go
func WithStackTrace(b ...bool) SpanEndEventOption {
	r := true
	for _, v := range b {
		r = v
	}
	return stackTraceOption(r)
}
```

Reply:
> Clear is better than clever

I do not like such `... hacks` - I even prefer ` WithStackTrace(b bool)`😉

Reply:
Yeah, I don't really like that use of variadic args either.  Clear is indeed better than clever.

Reply:
Thanks for suggestions. I have kept it like this `WithStackTrace(b bool)`